libvast/src/system/importer.cpp

```diff
@@ -270,9 +270,11 @@ importer(importer_actor::stateful_pointer<importer_state> self,
     if (self->state.stage) {
       self->state.stage->shutdown();
       self->state.stage->out().push(detail::framed<table_slice>::make_eof());
-      self->state.stage->out().force_emit_batches();
-      self->state.stage->out().close();
+      // We need to `fan_out_flush()` before we close, otherwise `close()`
+      // will delete all clean output paths and we might lose the eof.
```

Comment:
I think this also affects other places in the code. Should we create a function that wraps these three calls in proper order so that we only have a single place for this?

Reply:
Sounds reasonable, but I started implementing it and it quickly grew to a size where I think it makes more sense to make a topic branch with a separate review for this.